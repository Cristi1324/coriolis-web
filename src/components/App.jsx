/*
Copyright (C) 2017  Cloudbase Solutions SRL
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.
You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

// @flow

import React from 'react'
import { Switch, Route } from 'react-router-dom'
import styled, { injectGlobal } from 'styled-components'

import Fonts from './atoms/Fonts'
import Notifications from './organisms/Notifications'
import LoginPage from './pages/LoginPage'
import ReplicasPage from './pages/ReplicasPage'
import NotFoundPage from './pages/NotFoundPage'
import ReplicaDetailsPage from './pages/ReplicaDetailsPage'
import MigrationsPage from './pages/MigrationsPage'
import MigrationDetailsPage from './pages/MigrationDetailsPage'
import EndpointsPage from './pages/EndpointsPage'
import EndpointDetailsPage from './pages/EndpointDetailsPage'
import WizardPage from './pages/WizardPage'
import userStore from '../stores/UserStore'
import AssessmentsPage from './pages/AssessmentsPage'
import AssessmentDetailsPage from './pages/AssessmentDetailsPage'
import UsersPage from './pages/UsersPage'
import UserDetailsPage from './pages/UserDetailsPage'
import ProjectsPage from './pages/ProjectsPage'
import ProjectDetailsPage from './pages/ProjectDetailsPage'

import { navigationMenu } from '../constants'
import Palette from './styleUtils/Palette'
import StyleProps from './styleUtils/StyleProps'
import configLoader from '../utils/Config'

injectGlobal`
  ${Fonts}
  html, body, main {
    height: 100%;
  }
  body {
    margin: 0;
    color: ${Palette.black};
    font-family: Rubik;
    font-size: 14px;
    font-weight: ${StyleProps.fontWeights.regular};
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`
const Wrapper = styled.div`
  height: 100%;
  > div:first-child {
    height: 100%;
  }
`

type State = {
  isConfigReady: boolean,
}

class App extends React.Component<{}, State> {
  state = {
    isConfigReady: false,
  }

  componentWillMount() {
    userStore.tokenLogin()
    configLoader.load().then(() => {
      this.setState({ isConfigReady: true })
    })
  }

  render() {
    if (!this.state.isConfigReady) {
      return null
    }

    let renderOptionalPage = (name: string, component: any, path?: string, exact?: boolean) => {
      const isAdmin = userStore.loggedUser ? userStore.loggedUser.isAdmin : true
      let isDisabled = configLoader.config.disabledPages.find(p => p === name)
      if (navigationMenu.find(m => m.value === name && !isDisabled && (!m.requiresAdmin || isAdmin))) {
        return <Route path={`${path || `/${name}`}`} component={component} exact={exact} />
      }
      return null
    }

    return (
      <Wrapper>
        <Switch>
          <Route path="/" component={LoginPage} exact />
          <Route path="/login" component={LoginPage} />
          <Route path="/replicas" component={ReplicasPage} />
          <Route path="/replica/:id" component={ReplicaDetailsPage} exact />
          <Route path="/replica/:page/:id" component={ReplicaDetailsPage} />
          <Route path="/migrations" component={MigrationsPage} />
          <Route path="/migration/:id" component={MigrationDetailsPage} exact />
          <Route path="/migration/:page/:id" component={MigrationDetailsPage} />
          <Route path="/endpoints" component={EndpointsPage} />
          <Route path="/endpoint/:id" component={EndpointDetailsPage} />
          <Route path="/wizard/:type" component={WizardPage} />
          {renderOptionalPage('planning', AssessmentsPage)}
          {renderOptionalPage('planning', AssessmentDetailsPage, '/assessment/:info')}
          {renderOptionalPage('users', UsersPage)}
          {renderOptionalPage('users', UserDetailsPage, '/user/:id', true)}
          {renderOptionalPage('projects', ProjectsPage)}
          {renderOptionalPage('projects', ProjectDetailsPage, '/project/:id', true)}
          <Route component={NotFoundPage} />
        </Switch>
        <Notifications />
      </Wrapper >
    )
  }
}

export default App
