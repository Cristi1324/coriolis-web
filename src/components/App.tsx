/*
Copyright (C) 2017  Cloudbase Solutions SRL
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.
You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import { hot } from 'react-hot-loader/root'
import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import styled, { createGlobalStyle } from 'styled-components'
import { observe } from 'mobx'

import Fonts from './atoms/Fonts'
import Notifications from './organisms/Notifications'
import LoginPage from './pages/LoginPage'
import ReplicasPage from './pages/ReplicasPage'
import MessagePage from './pages/MessagePage'
import ReplicaDetailsPage from './pages/ReplicaDetailsPage'
import MigrationsPage from './pages/MigrationsPage'
import MigrationDetailsPage from './pages/MigrationDetailsPage'
import EndpointsPage from './pages/EndpointsPage'
import EndpointDetailsPage from './pages/EndpointDetailsPage'
import WizardPage from './pages/WizardPage'
import userStore from '../stores/UserStore'
import AssessmentsPage from './pages/AssessmentsPage'
import AssessmentDetailsPage from './pages/AssessmentDetailsPage'
import UsersPage from './pages/UsersPage'
import UserDetailsPage from './pages/UserDetailsPage'
import ProjectsPage from './pages/ProjectsPage'
import ProjectDetailsPage from './pages/ProjectDetailsPage'
import DashboardPage from './pages/DashboardPage'
import LogsPage from './pages/LogsPage'
import LogStreamPage from './pages/LogStreamPage'

import Tooltip from './atoms/Tooltip/Tooltip'

import { navigationMenu } from '../constants'
import Palette from './styleUtils/Palette'
import StyleProps from './styleUtils/StyleProps'
import configLoader from '../utils/Config'

const GlobalStyle = createGlobalStyle`
 ${Fonts}
  html, body, main {
    height: 100%;
    display: flex;
    flex-direction: column;
    min-height: 0;
  }
  body {
    margin: 0;
    color: ${Palette.black};
    font-family: Rubik;
    font-size: 14px;
    font-weight: ${StyleProps.fontWeights.regular};
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
`

const Wrapper = styled.div<any>`
  height: 100%;
  min-height: 0;
  display: flex;
  flex-direction: column;
  > div:first-child {
    height: 100%;
  }
`

type State = {
  isConfigReady: boolean,
}

class App extends React.Component<{}, State> {
  state = {
    isConfigReady: false,
  }

  awaitingRefresh: boolean = false

  async componentDidMount() {
    observe(userStore, 'loggedUser', () => {
      this.setState({})
    })
    await configLoader.load()
    userStore.tokenLogin()
    this.setState({ isConfigReady: true })
  }

  render() {
    if (!this.state.isConfigReady) {
      return null
    }

    const renderMessagePage = (options: {
      path: string,
      exact?: boolean,
      title: string,
      subtitle: string,
      showAuthAnimation?: boolean,
      showDenied?: boolean,
    }) => (
      <Route
        path={options.path}
        exact={options.exact}
        render={() => (
          <MessagePage
            title={options.title}
            subtitle={options.subtitle}
            showAuthAnimation={options.showAuthAnimation}
            showDenied={options.showDenied}
          />
        )}
      />
    )

    const renderRoute = (path: string, component: any, exact?: boolean) => {
      if (!userStore.loggedUser) {
        return renderMessagePage({
          path,
          exact,
          title: 'Authenticating...',
          subtitle: 'Please wait while authenticating user.',
          showAuthAnimation: true,
        })
      }
      return <Route path={path} component={component} exact={exact} />
    }

    const renderOptionalRoute = (name: string, component: any, path?: string, exact?: boolean) => {
      if (configLoader.config.disabledPages.find(p => p === name)) {
        return null
      }
      const actualPath = `${path || `/${name}`}`
      const requiresAdmin = Boolean(navigationMenu.find(n => n.value === name && n.requiresAdmin))
      if (!requiresAdmin) {
        return renderRoute(actualPath, component, exact)
      }
      if (!userStore.loggedUser || userStore.loggedUser.isAdmin == null) {
        return renderMessagePage({
          path: actualPath,
          exact,
          title: 'Checking permissions...',
          subtitle: 'Please wait while checking user\'s permissions.',
          showAuthAnimation: true,
        })
      }
      if (userStore.loggedUser && userStore.loggedUser.isAdmin === false) {
        return renderMessagePage({
          path: actualPath,
          exact,
          title: 'User doesn\'t have permissions to view this page',
          subtitle: 'Please login in with an administrator acount to view this page.',
          showDenied: true,
        })
      }
      if (userStore.loggedUser && userStore.loggedUser.isAdmin) {
        return <Route path={actualPath} exact={exact} component={component} />
      }
      return null
    }

    return (
      <Wrapper>
        <GlobalStyle />
        <Router>
          <Switch>
            {renderRoute('/', DashboardPage, true)}
            <Route path="/login" component={LoginPage} />
            {renderRoute('/dashboard', DashboardPage)}
            {renderRoute('/replicas', ReplicasPage)}
            {renderRoute('/replica/:id', ReplicaDetailsPage, true)}
            {renderRoute('/replica/:page/:id', ReplicaDetailsPage)}
            {renderRoute('/migrations', MigrationsPage)}
            {renderRoute('/migration/:id', MigrationDetailsPage, true)}
            {renderRoute('/migration/:page/:id', MigrationDetailsPage)}
            {renderRoute('/endpoints', EndpointsPage)}
            {renderRoute('/endpoint/:id', EndpointDetailsPage)}
            {renderRoute('/wizard/:type', WizardPage)}
            {renderOptionalRoute('planning', AssessmentsPage)}
            {renderOptionalRoute('planning', AssessmentDetailsPage, '/assessment/:info')}
            {renderOptionalRoute('users', UsersPage)}
            {renderOptionalRoute('users', UserDetailsPage, '/user/:id', true)}
            {renderOptionalRoute('projects', ProjectsPage)}
            {renderOptionalRoute('projects', ProjectDetailsPage, '/project/:id', true)}
            {renderOptionalRoute('logging', LogsPage)}
            {renderRoute('/streamlog', LogStreamPage)}
            <Route component={MessagePage} />
          </Switch>
        </Router>
        <Notifications />
        <Tooltip />
      </Wrapper>
    )
  }
}

export default hot(App)
