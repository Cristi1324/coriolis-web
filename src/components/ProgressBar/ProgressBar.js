/*
Copyright (C) 2017  Cloudbase Solutions SRL

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import React, { Component, PropTypes } from 'react';
import s from './ProgressBar.scss';
import withStyles from 'isomorphic-style-loader/lib/withStyles';


class ProgressBar extends Component {

  static propTypes = {
    progress: PropTypes.number
  }

  static defaultProps = {
    progress: 0
  }

  render() {
    let progressBarStyle = {
      width: this.props.progress + "%"
    }
    return <div className={s.root}>
      <div className={s.outer}>
        <div className={s.inner + " " + (this.props.progress == 100 ? s.completed : "")} style={progressBarStyle} />
      </div>
    </div>
  }

}

export default withStyles(ProgressBar, s);
