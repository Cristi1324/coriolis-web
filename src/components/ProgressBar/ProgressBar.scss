/*
Copyright (C) 2017  Cloudbase Solutions SRL

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

@import '../variables.scss';


.root {
  .outer {
    overflow: hidden;
    height: 8px;
    width: 100%;
    border: 1px solid $gray;
    border-radius: $border-radius;
    .inner {
      $bgWidth: 600px;
      height: 8px;
      transition: width $animation-swift-out;
      background-image: url('data:image/svg+xml;utf8,<svg width="$bgWidth" height="16px" viewBox="0 0 $bgWidth 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><defs><pattern id="pattern-1" width="12.4264706" height="12.4264706" x="-12.4264706" y="-12.4264706" patternUnits="userSpaceOnUse"><use xlink:href="%23image-2" transform="scale(0.155330882,0.155330882)"></use></pattern><image id="image-2" width="80" height="80" xlink:href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAFAAAABQCAYAAACOEfKtAAAABGdBTUEAALGPC/xhBQAAAeJJREFUeAHt3FtuwyAQBdCkq/EOmyWkO/RuWk+jtHlgDMMA87h8WUZ3BEdIlozxeV3X75NsuyzL8iVbkl9tm9/nlr7yK+STH/nu6t5QeKQjCRgOTxIwJJ4UYFg8CcDQeK2A4fFaAIFHelvjPIWBd7NjAQLvAY8ua1Yg8F7wagCBl8ArBQTeDl4JIPAyeEeAwDvAywECrwBvDxB4hXgpQOBV4L0CAq8S7xEQeAy8OyDwmHgUOzdkxaO9N4DEB3w6XdQAWsSj3UcVgFbxaEVPB7SMNx3QOt5UQA940wC94E0B9IQ3HNAb3lBAj3jDAL3iDQH0jNcd0DteV8AIeN0Ao+B1AYyEJw4YDU8UMCKeGGBUPBHAyHjNgNHxmgCBR3zMV/rAu+GxAIH3j1cNCLxnvCpA4L3jFQMCL41XBAi8fbxDQODl8bKAwDvG2wUEXhleEhB45XhvgMCrw3sCBF493h8g8Hh4v4DA4+NRko67dvspTdvQkml1H8TXnBdOzmjgTXV429yvVgBV4tHisQCoFs8CoGo87YDq8TQDmsDTCmgGTyOgKTxtgObwNAGaxNMCaBZPA6BpvNmA5vFmArrAmwXoBm8GoCu80YDu8EYCusQbBegWbwSga7zegO7xegKGwCPAH6jFkxG1FP01AAAAAElFTkSuQmCC"></image></defs><rect id="Base" stroke="none" fill="%230056B8" fill-rule="evenodd" x="0" y="0" width="$bgWidth" height="16"></rect><rect id="Base" stroke="none" fill="url(%23pattern-1)" fill-rule="evenodd" opacity="0.440443841" style="mix-blend-mode: luminosity;" x="0" y="0" width="$bgWidth" height="16"></rect></svg>');
      animation: bgMotion 1s infinite linear;
      &.completed {
        animation: none;
      }
    }
  }
}

@keyframes bgMotion {
  0% {
    background-position: -12px 0;
  }
  100% {
    background-position: 0 0;
  }
}

