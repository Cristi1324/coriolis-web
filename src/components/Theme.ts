/*
Copyright (C) 2021  Cloudbase Solutions SRL
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.
You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import { css } from 'styled-components'

const exactWidth = (width: string) => css`
  min-width: ${width};
  max-width: ${width};
`

const exactHeight = (height: string) => css`
  min-height: ${height};
  max-height: ${height};
`

export const ThemePalette = {
  primary: '#0044CB',
  primaryLight: '#000EA9',
  secondary: '#D9DCE3',
  secondaryLight: '#777A8B',
  black: '#202134',
  alert: '#F91661',
  success: '#4CD964',
  warning: '#FDC02F',
  grayscale: [
    '#D8DBE2', // 0
    '#ECEDF1', // 1
    '#C8CCD7', // 2
    '#A4AAB5', // 3
    '#616770', // 4
    '#7F8795', // 5
    '#1B2733', // 6
    '#F2F3F4', // 7
    '#858B93', // 8
  ],
}

export const ThemeProps = {
  fontWeights: {
    extraLight: 200,
    light: 300,
    regular: 400,
    medium: 500,
    bold: 600,
  },

  inputSizes: {
    regular: { width: 208, height: 32 },
    large: { width: 224, height: 32 },
    wizard: { width: 384 },
  },

  borderRadius: '4px',
  contentWidth: '928px',
  boxShadow: 'box-shadow: rgba(0, 0, 0, 0.1) 0 0 6px 2px;',

  animations: {
    swift: '.45s cubic-bezier(0.3, 1, 0.4, 1) 0s',
    rotation: css`
        animation: rotate 2s infinite linear;
        @keyframes rotate {
          from {transform: rotate(0deg);}
          to {transform: rotate(360deg);}
        }
      `,
    disabledLoading: css`
        animation: opacityToggle 750ms linear infinite alternate-reverse;
        @keyframes opacityToggle {
          0% {opacity: 1;}
          100% {opacity: 0.3;}
        }
      `,
  },

  mobileMaxWidth: 1350,

  exactWidth,
  exactHeight,
  exactSize: (size: string) => css`
      ${exactWidth(size)}
      ${exactHeight(size)}
    `,
}
