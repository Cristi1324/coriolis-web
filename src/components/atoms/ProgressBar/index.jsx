/*
Copyright (C) 2017  Cloudbase Solutions SRL
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.
You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

// @flow

import React from 'react'
import { observer } from 'mobx-react'
import styled from 'styled-components'

import Palette from '../../styleUtils/Palette'
import StyleProps from '../../styleUtils/StyleProps'

type Props = {
  progress: number,
  width?: number,
}

const Wrapper = styled.div`
  background: white;
`
const Progress = styled.div`
  height: 2px;
  background: ${Palette.primary};
  transition: all ${StyleProps.animations.swift};
  width: ${(props: Props) => props.width}%;
`

@observer
class ProgressBar extends React.Component<Props> {
  render() {
    return (
      <Wrapper {...this.props}>
        <Progress data-test-id="progressBar-progress" width={this.props.progress} />
      </Wrapper>
    )
  }
}

export default ProgressBar
