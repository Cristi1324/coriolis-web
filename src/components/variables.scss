/*
 * Colors
 * ========================================================================== */

$white-base:            hsl(255, 255, 255);
$gray-darker:           #1B2733;
$gray-dark:             #616770;
$gray:                  #A4AAB5;
$gray-light:            #ECEDF1;
$gray-lighter:          #D8DBE2;
$blue:                  #0056B8;
$blue-light:            #0068DD;
$black:                 #202134;
$red:                   #F70062;
$green:                 #34C555;

$color-replica:         #E62565;
$color-migration:       $blue;
/*
 * Typography
 * ========================================================================== */

$font-family-base:      'Rubik', sans-serif;

$weight-lightest: 100;
$weight-light: 300;
$weight-regular: 400;
$weight-semibold: 500;
$weight-medium: 600;


$header-text-color: $blue;
/*
 * Layout
 * ========================================================================== */

$max-content-width:     100%;
$narrow-content-width: 928px;
$wizard-content-width: 768px;

$border-radius: 4px;

/*
 * Media queries breakpoints
 * ========================================================================== */

$screen-xs-min:         480px;  /* Extra small screen / phone */
$screen-sm-min:         768px;  /* Small screen / tablet */
$screen-md-min:         992px;  /* Medium screen / desktop */
$screen-lg-min:         1200px; /* Large screen / wide desktop */

/*
 * Animations
 * ========================================================================== */

$animation-swift-out:   .45s cubic-bezier(0.3, 1, 0.4, 1) 0s;

