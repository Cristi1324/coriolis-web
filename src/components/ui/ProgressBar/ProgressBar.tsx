/*
Copyright (C) 2017  Cloudbase Solutions SRL
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.
You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

import React from "react";
import { observer } from "mobx-react";
import styled from "styled-components";

import { ThemePalette, ThemeProps } from "@src/components/Theme";

const Wrapper = styled.div`
  display: flex;
  align-items: center;
`;
const ProgressLabel = styled.div`
  text-align: right;
  min-width: 36px;
  font-size: 12px;
`;
const ProgressBarWrapper = styled.div`
  background: white;
  flex-grow: 1;
`;
const Progress = styled.div<{ width: number }>`
  height: 2px;
  background: ${ThemePalette.primary};
  transition: all ${ThemeProps.animations.swift};
  width: ${props => props.width}%;
`;
type Props = {
  progress: number;
  style?: React.CSSProperties;
  useLabel?: boolean;
};
@observer
class ProgressBar extends React.Component<Props> {
  render() {
    return (
      <Wrapper>
        <ProgressBarWrapper style={this.props.style}>
          <Progress width={this.props.progress} />
        </ProgressBarWrapper>
        {this.props.useLabel ? (
          <ProgressLabel>{this.props.progress} %</ProgressLabel>
        ) : null}
      </Wrapper>
    );
  }
}

export default ProgressBar;
