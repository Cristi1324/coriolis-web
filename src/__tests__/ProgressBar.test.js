import React from 'react';
import { ProgressBar } from '../components/ProgressBar';
import renderer from 'react-test-renderer';

describe('Progress Bar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ProgressBar />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders progress correctly', () => {
    const tree = renderer.create(
      <ProgressBar progress={20} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
})
