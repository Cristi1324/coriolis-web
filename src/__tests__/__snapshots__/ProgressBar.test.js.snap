// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Bar renders correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className="undefined "
      style={
        Object {
          "width": "0%",
        }
      }
    />
  </div>
</div>
`;

exports[`Progress Bar renders progress correctly 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className="undefined "
      style={
        Object {
          "width": "20%",
        }
      }
    />
  </div>
</div>
`;
